.github/workflows/generate-analysis.yml

```diff
@@ -0,0 +1,226 @@
+          install_wizard = env['upgrade.install.wizard'].create({})
+          install_wizard.select_odoo_modules()
+          install_wizard.install_modules()
+          env.cr.commit()
+          EOF
+          $ODOO_SHELL <<EOF
+          env['upgrade.generate.record.wizard'].create({}).generate()
+          env.cr.commit()
+          EOF
+      - name: Create current Odoo database
+        env:
+          ODOO: env-${{ env.TO_VERSION }}/bin/odoo -c odoo-${{ env.TO_VERSION }}.cfg -d ${{ env.TO_VERSION }}
+          ODOO_SHELL: env-${{ env.TO_VERSION }}/bin/odoo shell -c odoo-${{ env.TO_VERSION }}.cfg -d ${{ env.TO_VERSION }}
+        run: |
+          $ODOO --without-demo=all -i upgrade_analysis --stop-after-init
+          $ODOO_SHELL <<EOF
+          install_wizard = env['upgrade.install.wizard'].create({})
+          install_wizard.select_odoo_modules()
+          install_wizard.install_modules()
+          env.cr.commit()
+          EOF
+          $ODOO_SHELL <<EOF
+          env['upgrade.generate.record.wizard'].create({}).generate()
+          env.cr.commit()
+          EOF
+      - name: Delete all files to be generated from repository
+        env:
+          OPENUPGRADE_DIR: openupgrade-${{ env.TO_VERSION }}
+        run: |
+          for file in $OPENUPGRADE_DIR/openupgrade_scripts/scripts/*/*/{noupdate_changes.xml,upgrade_analysis.txt}; do
+            if [ ! -f $file ]; then
+              continue
+            fi
+            git -C $OPENUPGRADE_DIR rm ${file#$OPENUPGRADE_DIR/}
+          done
+      - name: Create analysis
+        env:
+          FROM_ODOO: env-${{ env.FROM_VERSION }}/bin/odoo -c odoo-${{ env.FROM_VERSION }}.cfg -d ${{ env.FROM_VERSION }}
+          TO_ODOO_SHELL: env-${{ env.TO_VERSION }}/bin/odoo shell -c odoo-${{ env.TO_VERSION }}.cfg -d ${{ env.TO_VERSION }}
+        run: |
+          $FROM_ODOO --max-cron-threads=0 &
+          ODOO_PID=$!
+          $TO_ODOO_SHELL <<EOF
+          config = env['upgrade.comparison.config'].create({
+              'database': '${{ env.FROM_VERSION }}',
+          })
+          analysis_wizard = env['upgrade.analysis'].create({
+              'config_id': config.id,
+          })
+          analysis_wizard._compute_upgrade_path()
+          analysis_wizard.analyze()
+          env.cr.commit()
+          print(analysis_wizard.log)
+          EOF
+          kill -s KILL $ODOO_PID
+      - name: Determine changed addons
+        id: generate_body
+        env:
+          OPENUPGRADE_DIR: openupgrade-${{ env.TO_VERSION }}
+          PR_BODY: pr_body.txt
+        run: |
+          git -C $OPENUPGRADE_DIR add .
+
+          echo "Analysis or noupdate changes for modules marked as done:
+
+          " > "$PR_BODY"
+
+          for module in $(
+            git -C $OPENUPGRADE_DIR status --short | awk '{print $2}' | awk -F/ '{print $3}' | sort -u
+          ); do
+            if grep -qE "\<$module\>.*\<((Done)|(No))" $OPENUPGRADE_DIR/docsource/modules*; then
```

Comment:
When new analysis is done, it deletes all the content in Status column of docsource, so you will not detect anything here if I understood correctly your code.

Reply:
this looks at the rst file, which this code doesn't touch at all. Check out https://github.com/hbrunn/OpenUpgrade/pull/16 where finds some modules mentioned there